### PR TITLE
feat: runtime

### PR DESCRIPTION
**Pull request by anchorgoogle**

### CODE REVIEW

File: src/compute/runtime/app/controller/vm.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
+//@ts-nocheck
+import { Controller } from '../../src/base/controller';
+import { bp } from '../../src/blueprint';
+import { randomBytes, scrypt, createHmac } from "crypto";
+
+const fs = require('fs');
```
Comment (blaesus) [inline src/compute/runtime/app/controller/vm.ts:6]: Better use `import * as fs from "fs"` for typed imports.

File: src/compute/runtime/app/controller/vm.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    @bp.get('/vm/:project/:fn')
+    @bp.get('/vm/:project/:fn/:param')
+    @bp.post('/vm/:project/:fn')
+    @bp.post('/vm/:project/:fn/:param')
+    @bp.del('/vm/:project/:fn')
+    @bp.del('/vm/:project/:fn/:param')
+    async index() {
+        let timr = null;
```
Comment (blaesus) [inline src/compute/runtime/app/controller/vm.ts:87]: timer?

File: src/compute/runtime/app/controller/vm.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
+//@ts-nocheck
```
Comment (blaesus) [inline src/compute/runtime/app/controller/vm.ts:1]: Let's try to avoid disabling TypeScript...

File: src/compute/runtime/src/blueprint.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+const methods = ['get', 'post', 'patch', 'del', 'options', 'put']
+
+interface Bp {
+    [key: string]: Array<{
+        httpMethod: string,
+        constructor: any,
+        handler: string
+    }>
+}
+interface BP {
```
Comment (blaesus) [inline src/compute/runtime/src/blueprint.ts:12]: Can we give `Bp` and `BP` more meaningful names? They're quite confusing at the moment.

File: src/compute/runtime/src/blueprint.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    patch(url: string): Decorator;
+    del(url: string): Decorator;
+    options(url: string): Decorator;
+    put(url: string): Decorator;
+}
+
+class Blueprint {
+    router: Bp = {}
+    setRouter(url: string, blueprint: BP) {
+        const _bp = this.router[url];
```
Comment (blaesus) [inline src/compute/runtime/src/blueprint.ts:42]: `for bp of this.router[url]`?

**Review by blaesus — CHANGES_REQUESTED** [review]

File: src/compute/runtime/app/controller/vm.ts (hunk 3 above)
Comment (anchorgoogle) [reply src/compute/runtime/app/controller/vm.ts]: I was eager to verify, so I banned it and forgot to open it

File: src/compute/runtime/src/blueprint.ts (hunk 4 above)
Comment (anchorgoogle) [reply src/compute/runtime/src/blueprint.ts]: It becomes the definition of routing, the meaning is clearer

File: src/compute/runtime/src/blueprint.ts (hunk 5 above)
Comment (anchorgoogle) [reply src/compute/runtime/src/blueprint.ts]: It means routing information group

**Review by blaesus — APPROVED** [review]
Looks good. Let's merge.

**Pull request merged**